### PR TITLE
Performance

### LINKED ISSUE #1611: Improve app performance
Xamarin.Forms has [this doc](https://learn.microsoft.com/en-us/xamarin/xamarin-forms/deploy-test/performance) about improving app performance. Migrate/update this doc to .NET MAUI, removing items that are no longer accurate.

There will be additional items to add to this doc that will be tracked elsewhere.

---
[Associated WorkItem - 143151](https://dev.azure.com/msft-skilling/Content/_workitems/edit/143151)

### PR DESCRIPTION
**Pull request by davidbritch**
Fixes #1611 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/deployment/index.md](https://github.com/dotnet/docs-maui/blob/9a0397f23d36a29f33b64faa42cd481870371a34/docs/deployment/index.md) | [Deployment](https://review.learn.microsoft.com/en-us/dotnet/maui/deployment/index?branch=pr-en-us-1622) |
| [docs/deployment/performance.md](https://github.com/dotnet/docs-maui/blob/9a0397f23d36a29f33b64faa42cd481870371a34/docs/deployment/performance.md) | [docs/deployment/performance](https://review.learn.microsoft.com/en-us/dotnet/maui/deployment/performance?branch=pr-en-us-1622) |

### CODE REVIEW

File: docs/deployment/performance.md [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+        <Label Grid.Row="3"
+               Text="Address:" />
+        <Entry Grid.Row="3"
+               Grid.Column="1"
+               Placeholder="Enter your address" />
+    </Grid>
+</ContentPage>
+```
+
+## Optimize layout performance
```
Comment (davidbritch) [inline docs/deployment/performance.md]: @hartez Is this section accurate/worth keeping?

File: docs/deployment/performance.md [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+- Return a `Task` object, instead of returning an awaited `Task` object, when there's no need to process the result of an asynchronous operation. This is more performant due to less context switching being performed.
+- Use the Task Parallel Library (TPL) Dataflow library in scenarios such as processing data as it becomes available, or when you have multiple operations that must communicate with each other asynchronously. For more information, see [Dataflow (Task Parallel Library)](/dotnet/standard/parallel-programming/dataflow-task-parallel-library).
+
+### UI
+
+The following guidelines should be followed when using the TAP with UI controls:
+
+- Call an asynchronous version of an API, if it's available. This will keep the UI thread unblocked, which will help to improve the user's experience with the app.
+- Update UI elements with data from asynchronous operations on the UI thread, to avoid exceptions being thrown. However, updates to the `ListView.ItemsSource` property will automatically be marshaled to the UI thread. For information about determining if code is running on the UI thread, see [Create a thread on the UI thread](~/platform-integration/appmodel/main-thread.md).
```
Comment (davidbritch) [inline docs/deployment/performance.md:259]: @hartez Is this still accurate?

Comment (hartez) [reply docs/deployment/performance.md:259]: Yes, the ListView will still auto-dispatch the changes to the main thread.

File: docs/deployment/performance.md (hunk 1 above)
Comment (hartez) [reply docs/deployment/performance.md]: Line 122 isn't always true, and can probably be dropped.

**Review by hartez — APPROVED** [review]

**Comment by davidbritch** [comment]
Replace markdown xrefs with XML xrefs.

**Pull request merged**